libs/application/templates/passport-annulment/src/fields/PassportSelection/index.tsx

```diff
@@ -0,0 +1,181 @@
+  } = useFormContext()
+  const userPassportRadio = `${id}.userPassport`
+  const childPassportRadio = `${id}.childPassport`
+  const fieldErros = getErrorViaPath(errors, userPassportRadio)
+  const identityDocumentData = application.externalData.identityDocument
+    .data as IdentityDocumentData
+
+  const tag = (identityDocument: IdentityDocument) => {
+    const today = new Date()
+    const expirationDate = new Date(identityDocument?.expirationDate)
+    const todayPlus6Months = new Date(
+      new Date().setMonth(new Date().getMonth() + 6),
+    )
+
+    let tagObject = {} as Tag
+
+    if (!identityDocument) {
+      tagObject = {
+        label: formatMessage(m.noPassport),
+        variant: 'blue',
+        outlined: true,
+      }
+    } else if (today > expirationDate) {
+      tagObject = {
+        label: formatMessage(m.expiredTag),
+        variant: 'red',
+        outlined: true,
+      }
+    } else if (todayPlus6Months > expirationDate) {
+      tagObject = {
+        label:
+          formatMessage(m.validTag) +
+          ' ' +
+          (identityDocument
+            ? format(new Date(expirationDate), 'dd/MM/yy')
+            : ''),
+        variant: 'mint',
+        outlined: true,
+      }
+    } else if (todayPlus6Months < expirationDate) {
+      tagObject = {
+        label: formatMessage(m.validTag) + ' ' + expirationDate.getFullYear(),
+        variant: 'mint',
+        outlined: true,
+      }
+    }
+
+    return tagObject
+  }
+  return (
+    <Box>
+      <RadioFormField
+        error={fieldErros}
+        application={application}
+        field={{
+          id: userPassportRadio,
+          title: '',
+          type: FieldTypes.RADIO,
+          component: FieldComponents.RADIO,
+          children: undefined,
+          backgroundColor: 'white',
+          defaultValue: '',
+          options: [
+            {
+              label: (application.externalData.nationalRegistry.data as any)
+                ?.fullName,
+              value:
+                identityDocumentData.userPassport?.productionRequestID +
+                ',' +
+                identityDocumentData.userPassport?.numberWithType +
+                ',' +
+                (application.externalData.nationalRegistry.data as any)
+                  ?.fullName,
+              subLabel: identityDocumentData.userPassport
+                ? formatMessage(m.passportNumber) +
+                  ' ' +
+                  identityDocumentData.userPassport?.numberWithType
+                : '',
+              tag: tag(identityDocumentData.userPassport),
+              disabled:
+                tag(identityDocumentData.userPassport).variant !== 'mint',
+            },
+          ],
+          onSelect: (e) => {
+            setValue('productionRequestID', e.split(',')[0])
+            setValue('passportNumber', e.split(',')[1])
+            setValue('passportName', e.split(',')[2])
+            setValue(childPassportRadio, '')
+          },
```

Comment:
Random nitpick. Is something like this cleaner?
```typescript
          onSelect: (e) => {
            let [productionRequestID, passportNumber, passportName] = e.split(',')
            setValue('productionRequestID', productionRequestID)
            setValue('passportNumber', passportNumber)
            setValue('passportName', passportName)
            setValue(childPassportRadio, '')
          },
```

Reply:
Especially if we add on to this functionality, it's nice to have some semantics around the positions of the split?